<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>OCC::KillSwitchSettings</class>
 <widget class="QWidget" name="OCC::KillSwitchSettings">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>500</width>
    <height>650</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Kill Switch Protection</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="statusGroup">
     <property name="title">
      <string>Protection Status</string>
     </property>
     <layout class="QHBoxLayout" name="statusLayout">
      <item>
       <widget class="QLabel" name="statusIndicator">
        <property name="minimumSize">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
        <property name="styleSheet">
         <string>background-color: #00FF00; border-radius: 10px;</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="statusLabel">
        <property name="text">
         <string>Protected - No Threats</string>
        </property>
        <property name="styleSheet">
         <string>font-weight: bold;</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="statusSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="resetButton">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>Reset Kill Switch</string>
        </property>
        <property name="styleSheet">
         <string>QPushButton { background-color: #FF6B6B; color: white; font-weight: bold; padding: 5px 15px; }</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="enableKillSwitch">
     <property name="text">
      <string>Enable Kill Switch Protection</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
     <property name="styleSheet">
      <string>font-weight: bold; font-size: 14px;</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="settingsGroup">
     <property name="title">
      <string>Detection Settings</string>
     </property>
     <layout class="QFormLayout" name="settingsLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="deleteThresholdLabel">
        <property name="text">
         <string>Mass deletion threshold:</string>
        </property>
        <property name="toolTip">
         <string>Trigger kill switch when this many files are deleted</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <layout class="QHBoxLayout" name="deleteLayout">
        <item>
         <widget class="QSpinBox" name="deleteThresholdSpinBox">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>1000</number>
          </property>
          <property name="value">
           <number>10</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="deleteThresholdSuffix">
          <property name="text">
           <string>files</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="timeWindowLabel">
        <property name="text">
         <string>Time window:</string>
        </property>
        <property name="toolTip">
         <string>Time period to count deletions</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <layout class="QHBoxLayout" name="timeLayout">
        <item>
         <widget class="QSpinBox" name="timeWindowSpinBox">
          <property name="minimum">
           <number>10</number>
          </property>
          <property name="maximum">
           <number>3600</number>
          </property>
          <property name="value">
           <number>60</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="timeWindowSuffix">
          <property name="text">
           <string>seconds</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="entropyThresholdLabel">
        <property name="text">
         <string>Entropy threshold:</string>
        </property>
        <property name="toolTip">
         <string>Files with entropy above this are considered encrypted (max 8.0)</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <layout class="QHBoxLayout" name="entropyLayout">
        <item>
         <widget class="QDoubleSpinBox" name="entropyThresholdSpinBox">
          <property name="minimum">
           <double>5.000000000000000</double>
          </property>
          <property name="maximum">
           <double>8.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>7.500000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="entropyThresholdSuffix">
          <property name="text">
           <string>bits/byte</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="canaryGroup">
     <property name="title">
      <string>Canary Files (Honeypots)</string>
     </property>
     <layout class="QVBoxLayout" name="canaryLayout">
      <item>
       <widget class="QLabel" name="canaryDescription">
        <property name="text">
         <string>Canary files are decoy files that should never be modified. Any change triggers immediate protection.</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
        <property name="styleSheet">
         <string>color: gray; font-style: italic;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QListWidget" name="canaryListWidget">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>100</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="canaryButtonLayout">
        <item>
         <widget class="QPushButton" name="addCanaryButton">
          <property name="text">
           <string>Add...</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="removeCanaryButton">
          <property name="text">
           <string>Remove</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="canaryButtonSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="historyGroup">
     <property name="title">
      <string>Threat History</string>
     </property>
     <layout class="QVBoxLayout" name="historyLayout">
      <item>
       <widget class="QListWidget" name="threatListWidget">
        <property name="styleSheet">
         <string>QListWidget { font-family: monospace; }</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="historyButtonLayout">
        <item>
         <spacer name="historyButtonSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="clearHistoryButton">
          <property name="text">
           <string>Clear History</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
